docs/content/release-notes/flink-1.15.md

```diff
@@ -0,0 +1,379 @@
+  (FLINK-24687). Besides, the artifactId of `flink-orc`, `flink-orc-nohive`, 
+  `flink-parquet` has no Scala suffix (\_2.11 / \_2.12) anymore.
+
+## JDK Upgrade
+
+The support of Java 8 is now deprecated and will be removed in a future release 
+([FLINK-25247](https://issues.apache.org/jira/browse/FLINK-25247)). We recommend 
+all users to migrate to Java 11.
+
+The default Java version in the Flink docker images is now Java 11 
+([FLINK-25251](https://issues.apache.org/jira/browse/FLINK-25251)). 
+There are images built with Java 8, tagged with “java8”. 
+
+## DataStream API
+
+### [TypeSerializer version mismatch during eagerly restore (FLINK-24858)](https://issues.apache.org/jira/browse/FLINK-24858)
+
+This ticket resolves an issue that during state migration between Flink versions
+the wrong serializer might have been picked.
+
+When upgrading from Flink 1.13.x please immediately choose 1.14.3 or higher and 
+skip 1.14.0, 1.14.1, 1.14.2 because all are affected and it might prevent your 
+job from starting.
+
+## Table API & SQL
+
+### [Make the legacy behavior disabled by default (FLINK-26551)](https://issues.apache.org/jira/browse/FLINK-26551)
+
+The legacy casting behavior has been disabled by default. This might have 
+implications on corner cases (string parsing, numeric overflows, to string 
+representation, varchar/binary precisions). Set 
+`table.exec.legacy-cast-behaviour=ENABLED` to restore the old behavior.
+
+### [Enforce CHAR/VARCHAR precision when outputting to a Sink (FLINK-24753)](https://issues.apache.org/jira/browse/FLINK-24753)
+
+`CHAR`/`VARCHAR` lengths are enforced (trimmed/padded) by default now before entering
+the table sink.
+
+### [Support the new type inference in Scala Table API table functions (FLINK-26518)](https://issues.apache.org/jira/browse/FLINK-26518)
+
+Table functions that are called using Scala implicit conversions have been updated 
+to use the new type system and new type inference. Users are requested to update 
+their UDFs or use the deprecated `TableEnvironment.registerFunction` to restore 
+the old behavior temporarily by calling the function via name.
+
+### [Propagate executor config to TableConfig (FLINK-26421)](https://issues.apache.org/jira/browse/FLINK-26421)
+
+`flink-conf.yaml` and other configurations from outer layers (e.g. CLI) are now 
+propagated into `TableConfig`. Even though configuration set directly in `TableConfig` 
+has still precedence, this change can have side effects if table configuration 
+was accidentally set in other layers.
+
+### [Remove pre FLIP-84 methods (FLINK-26090)](https://issues.apache.org/jira/browse/FLINK-26090)
+
+The previously deprecated methods `TableEnvironment.execute`, `Table.insertInto`,
+`TableEnvironment.fromTableSource`, `TableEnvironment.sqlUpdate`, and 
+`TableEnvironment.explain` have been removed. Please use the provided 
+alternatives introduced in 
+[FLIP-84](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=134745878).
```

Comment:
IMO we shouldn't link to FLIPs in user-facing content. If we want to refer them to anything, then I guess it should be the javadocs that should contain migration instructions.

Reply:
changed the text, will also add some javadocs links.